### DISCUSSION TITLE
Prevent Page.save() from updating all descendant URLs

### DISCUSSION THREAD

**Discussion by tian-yan** [wagtail/wagtail#7686] [General]
I have a blog like wagtail site, integrate with **`django_comments_xtd`** for commenting purposes.
each blog url format: https://domain.com/blogs/<blog-id\>/
comments associated with above blog url format: https://domain.com/blogs/<blog-id\>/c<comment-id\>/
It is clear that **comments' url become the descendant urls of the related blog post url**. 

#### [wagtail/core/models/__init__.py](https://github.com/wagtail/wagtail/blob/d8e7fdf1ee96e3b1353d505ed83d0a1416c235e9/wagtail/core/models/__init__.py#L463)
```
class Page(AbstractPage, index.Indexed, ClusterableModel, metaclass=PageBase):
    ...

    @transaction.atomic
    # ensure that changes are only committed when we have updated all descendant URL paths, to preserve consistency
    def save(self, clean=True, user=None, log_action=False, **kwargs):
        """
        Overrides default method behaviour to make additional updates unique to pages,
        such as updating the ``url_path`` value of descendant page to reflect changes
        to this page's slug.
        New pages should generally be saved via the ``add_child()`` or ``add_sibling()``
        method of an existing page, which will correctly set the ``path`` and ``depth``
        fields on the new page before saving it.
        By default, pages are validated using ``full_clean()`` before attempting to
        save changes to the database, which helps to preserve validity when restoring
        pages from historic revisions (which might not necessarily reflect the current
        model state). This validation step can be bypassed by calling the method with
        ``clean=False``.
        """
```
According to above wagtail source code, when I revise existing blog post's content **(not slug)**, **`page.save()`** will be triggerred and because of **updating all descendant urls**, all existing associated comments for that blog will be **saved** again, in other words, **`comment.save()`** will be triggerred which is verified in my debug.  This is not reasonable, revising existing blog post's contents **(not slug)** should not trigger **`save()`** method of existing comments. 

**Q: How can I achieve my intention that revising existing blog post's contents **(not slug)** will never trigger **`save()`** method of comments ?**

**Comment by cnk** [comment]
Why do you need to prevent triggering the comment's `save` method? If the slug doesn't change, then the comment save should be a `no-op`. Are you seeing changes to your comments when changing content on your blog post? If so, what?
